src/lib/storage/value_segment.cpp

```diff
@@ -6,68 +6,84 @@
 namespace opossum {
 
 template <typename T>
-ValueSegment<T>::ValueSegment(bool nullable) {
-  // Implementation goes here
-}
+ValueSegment<T>::ValueSegment(bool nullable) : _is_nullable(nullable) {}
 
 template <typename T>
 AllTypeVariant ValueSegment<T>::operator[](const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (is_null(chunk_offset)) {
+    return NULL_VALUE;
+  }
+
+  return _values.at(chunk_offset);
 }
 
 template <typename T>
 bool ValueSegment<T>::is_null(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (!is_nullable()) {
+    return false;
+  }
+
+  return _nulls.at(chunk_offset);
```

Comment:
You could "simplify": 
```suggestion
   return is_nullable() && _nulls.at(chunk_offset);
```
the second condition isn't evaluated, if is_nullable is already false because of [short circuit evaluation](https://stackoverflow.com/questions/5211961/how-does-c-handle-short-circuit-evaluation#:~:text=Yes%2C%20the%20%26%26,and%20%7C%20operators.)